universal.md

```diff
@@ -0,0 +1,423 @@
+			}
+		},
+		{
+			"rtype": "reference",
+			"descriptor": {
+  				"mediaType": "application/vnd.oci.object.manifest.v1+json",
+  				"size": 1682,
+ 				"digest": "sha256:3b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270"
+			}
+		}
+		]
+	}
+	]
+}
+```
+
+Now let us look at how this would map the following image manifest
+```
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+    "size": 7023,
+    "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7"
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 32654,
+      "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 16724,
+      "digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 73109,
+      "digest": "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736"
+    }
+  ],
+  "annotations": {
+    "com.example.key1": "value1",
+    "com.example.key2": "value2"
+  }
+}
+```
+Note that we should get clients to support this generic manifest for images too so we can transition and support upgradeability.
+```
+{
+	"schemaVersion": "1",
+	"mediaType": "application/vnd.oci.object.manifest.v1+json",
+	"objects": [
+	{
+		"type": "org.oci.image",
+		"version": "2",
+		"components": [
+		{
+			"rtype": "blob",
+			"ctype": "org.oci.config",
+			"descriptor": {
+			   "mediaType": "application/vnd.oci.image.config.v1+json",
+    			"size": 7023,
+    			"digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7"
+  			},
+		},
+		{
+			"rtype": "blob",
+			"ctype": "org.oci.layer",
+			"descriptor": {
+    			"mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      			"size": 32654,
+      			"digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0"
+  			}
+		},
+		{
+			"rtype": "blob",
+			"ctype": "org.oci.layer",
+			"descriptor": {
+    			"mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      			"size": 16724,
+      			"digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b"
+  			}
+		},
+		{
+			"rtype": "blob",
+			"ctype": "org.oci.layer",
+			"descriptor": {
+    			"mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      			"size": 73109,
+      			"digest": "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736"
+    		}
+		},
+		]
+	}
+	],
+	"annotations": {
+    	"com.example.key1": "value1",
+    	"com.example.key2": "value2"
+  }
+}
+```
+If we want to add an SBOM to this image we can add a Property, like a pointer but pointing at the object itself.
```

Comment:
I finally got to drawing this out. I am still confused about how embedding a property in the object manifest is different from creating a relation manifest. Does it have to do with upgradability?

Reply:
> got to drawing this out

Do you mean literally drawing something? Can I see it? 😄

Reply:
<img width="1562" alt="Screen Shot 2021-06-04 at 9 16 37 AM" src="https://user-images.githubusercontent.com/1007786/120832633-ba914600-c515-11eb-9a4b-d6be0f58e810.png">

Nothing fancy but helps some...

Reply:
I think you would need a third box for the SBOM that both the `sha256:3b0b...` descriptors point to, because as shown this has an impossible cycle.

Reply:
We iterated a bit on this, and here's a PR on the PR that attempted to incorporate SBoMs and some name cleanup: https://github.com/justincormack/artifacts/pull/1
Here are some examples of how an SBoM could be associated: https://github.com/SteveLasker/artifacts/blob/distribution-manifest/universal.md#sbom-reference-artifact

The rest of the manifest examples are at: https://github.com/SteveLasker/artifacts/tree/distribution-manifest/manifest-examples

Justin had concerns of cycles due to the collections, which is what led to the change of a single `subjectManifest` in #29

Reply:
Sorry, I had a hard time following all the proposed manifests. I suppose part of the issue is that the OCI descriptor is a complex type itself. If y'all would like, I can make more drawings for these proposals...

Reply:
> If y'all would like, I can make more drawings for these proposals...

I make drawings to learn as well, so I'm sure it would be helpful to others. It would also be nice to validate that we're all on the same page re: what's actually being proposed, but don't feel pressured to if you're busy :P